src/app/rundown-view/components/rundown-view/rundown-view.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, OnDestroy, OnInit } from '@angular/core'
+import { Rundown } from '../../../core/models/rundown'
+import { SubscriptionLike } from 'rxjs'
+import { RundownStateService } from '../../../core/services/rundown-state.service'
+import { ActivatedRoute } from '@angular/router'
+import { Logger } from '../../../core/abstractions/logger.service'
+
+@Component({
+  selector: 'sofie-rundown-view',
+  templateUrl: './rundown-view.component.html',
+  styleUrls: ['./rundown-view.component.scss'],
+})
+export class RundownViewComponent implements OnInit, OnDestroy {
+  public rundown?: Rundown
+  private rundownSubscription?: SubscriptionLike
+  private readonly logger: Logger
+
+  constructor(
+    private readonly route: ActivatedRoute,
+    private readonly rundownStateService: RundownStateService
+  ) {}
+
+  public ngOnInit(): void {
+    const rundownId: string | null = this.route.snapshot.paramMap.get('rundownId')
+    if (!rundownId) {
+      this.logger.error("[error]: No rundownId found. Can't fetch Rundown")
+      return
+    }
+    this.rundownStateService
+      .subscribeToRundown(rundownId, rundown => {
+        this.rundown = rundown
+      })
+      .then(unsubscribeFromRundown => {
+        this.rundownSubscription = unsubscribeFromRundown
+      })
+      .catch(error => this.logger.error(`[error] Failed subscribing to rundown with id '${rundownId}'.`, error))
```

Comment:
The logger is used incorrectly here. You can use the `.data(data: unknown)` method prior to calling `.error`, such that this becomes:
```suggestion
      .catch(error => this.logger.data(error).error(`[error] Failed subscribing to rundown with id '${rundownId}'.`))
```

The second parameter of `.error` should be a metadata object that will be spread out into the root of the log entry. The same structure that is the outcome of the suggestion, can be achieved by wrapping the error in an object like so:
```suggestion
      .catch(error => this.logger.error(`[error] Failed subscribing to rundown with id '${rundownId}'.`, { error }))
```

Reply:
I have used the suggestion!